examples/cosmos/ics20/ics20.qnt

```diff
@@ -0,0 +1,129 @@
+// -*- mode: Bluespec; -*-
+
+/**
+ * A specification ICS20 fungible token transfer protocol.
+ *
+ * For now, this only models the denomination trace changes between transfers.
+ *
+ * Gabriela Moreira, Informal Systems, 2023
+ */
+module ics20 {
+  type Uint = int
+
+  type State = {
+    trace: List[str],
+    denom: str,
+  }
+
+  type FungibleTokenPacketData = {
+    trace: List[str],
+    denom: str,
+    amount: Uint,
+    sender: str,
+    receiver: str,
+  }
+
+  type Packet = {
```

Comment:
I really wonder whether we should remove the concept of a packet from the spec of denominations. I find IBC denominations confusing enough, so we don't have to add packets in the picture.